tests/test_runner.py

```diff
@@ -0,0 +1,166 @@
+            cls.test_xml = fileh.read()
+
+    def test_getrunner(self):
+        """Ensure getrunner() can create a runner subclass"""
+        result = runner.getrunner('beaker', {'jobtemplate': 'test'})
+        self.assertIsInstance(result, runner.BeakerRunner)
+
+    def test_invalid_getrunner(self):
+        """Ensure getrunner() throws an exception for an invalid runner"""
+        bad_runner = 'pizza'
+        with self.assertRaises(Exception) as context:
+            runner.getrunner(bad_runner, {'jobtemplate': 'test'})
+
+        self.assertTrue(
+            "Unknown runner type: {}".format(bad_runner)
+            in context.exception
+        )
+
+    def test_getxml(self):
+        """Ensure BeakerRunner.getxml() returns xml"""
+        myrunner = runner.BeakerRunner(**DEFAULT_ARGS)
+        result = myrunner.getxml({})
+        self.assertEqual(result, self.test_xml)
+
+    def test_getxml_replace(self):
+        """Ensure BeakerRunner.getxml() returns xml with replacements"""
+        myrunner = runner.BeakerRunner(**DEFAULT_ARGS)
+        result = myrunner.getxml({'TEST': 'replaced text'})
+        expected_xml = self.test_xml.replace("##TEST##", "replaced text")
+        self.assertEqual(result, expected_xml)
+
+    @mock.patch('subprocess.Popen')
+    def test_getresultstree(self, mock_popen):
+        """Ensure getresultstree() works"""
+        test_xml = "<xml><test>TEST</test></xml>"
+        mock_popen.return_value.returncode = 0
+        mock_popen.return_value.communicate.return_value = (test_xml, '')
+        myrunner = runner.BeakerRunner(**DEFAULT_ARGS)
+        result = myrunner.getresultstree(jobid=0)
+        self.assertEqual(next(x.text for x in result.iter('test')), 'TEST')
+
+    @mock.patch('subprocess.Popen')
+    def test_dumpjunitresults(self, mock_popen):
+        """Ensure dumpjunitresults() works"""
+        mock_popen.return_value.returncode = 0
+        mock_popen.return_value.communicate.return_value = (
+            'stdout',
+            'stderr'
+        )
+
+        junit_dir = tempfile.mkdtemp()
+        myrunner = runner.BeakerRunner(**DEFAULT_ARGS)
+        myrunner.dumpjunitresults("J:00001", junit_dir)
+        expected_file = "{}/j_00001.xml".format(junit_dir)
+        self.assertTrue(os.path.exists(expected_file))
+
+    @mock.patch('skt.runner.BeakerRunner.getresultstree')
+    def test_getconsolelog(self, mock_getresultstree):
+        """Ensure getconsolelog() works"""
+        # Mock up a beaker XML reply
+        beaker_xml = '{}/assets/beaker_results.xml'.format(SCRIPT_PATH)
+        with open(beaker_xml, 'r') as fileh:
+            mocked_xml = fileh.read()
+        mock_getresultstree.return_value = etree.fromstring(mocked_xml)
+
+        myrunner = runner.BeakerRunner(**DEFAULT_ARGS)
+        result = myrunner.getconsolelog()
+        self.assertEqual(result, "http://example.com/")
+
+    def test_forget_cid_withj(self):
+        """Ensure _forget_cid() works with J type jobs"""
```

Comment:
The `bkr` manpage [describes the terminology](https://www.mankier.com/1/bkr#Description-Specifying_tasks) applicable here. I'm not sure what Artem meant by "cid", perhaps "check ID" (would be good to find out). However, outside of those we need to use Beaker terminology (i.e. "taskspec", "job", "recipe set", "recipe", etc.), not "J type jobs" or "R type jobs". We need to stay consistent, so that the code is easier to link to Beaker documentation and Web UI and thus easier to understand. Please fix. Later we might need to rename `cid` as well.

Also, please document the `_forget_cid` function before testing it.